styx-common/src/main/java/com/spotify/styx/api/Middlewares.java

```diff
@@ -231,10 +232,14 @@ private static AuthContext auth(RequestContext requestContext) {
     });
   }
 
-  private static <T> Middleware<AsyncHandler<Response<T>>, AsyncHandler<Response<T>>> authValidator() {
+  public static <T> Middleware<AsyncHandler<Response<T>>, AsyncHandler<Response<T>>> authValidator() {
     return h -> rc -> {
-      // ensure auth context is valid (throws otherwise)
-      auth(rc);
+      if (!"GET".equals(rc.request().method())) {
+        if (!auth(rc).user().isPresent()) {
+          return completedFuture(
+              Response.forStatus(Status.UNAUTHORIZED.withReasonPhrase("Unauthorized access")));
```

Comment:
Not really. Normally returning a future is needed, and in this case it might be just to make `auth` easier to call and handle.